Panic when declaring a variable of an unknown type

```
package main
func main()() {
    var i foo
}
```
Output:
```
panic: type 'foo' does not exist

goroutine 1 [running, locked to thread]:
github.com/skycoin/cx/cxgo/actions.DeclarationSpecifiersStruct(0xc4201e87e0, 0x3, 0x0, 0x0, 0xc4200c4d00, 0xc4201a3c01)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/actions/declarations.go:359 +0x6d3
github.com/skycoin/cx/cxgo/cxgo0.yyParse(0x973800, 0xc42009b040, 0xc42009b040)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo0/cxgo0.go:1910 +0x1ed7
github.com/skycoin/cx/cxgo/cxgo0.Parse(0xc4200902a0, 0x2e, 0x2d)
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo0/cxgo0.go:74 +0xc8
main.main()
	/home/reash/go/src/github.com/skycoin/cx/cxgo/cxgo.nn.go:8556 +0x367a
```

--- a/cx/execute.go
+++ b/cx/execute.go
@@ -169,12 +169,12 @@ func (prgrm *CXProgram) RunCompiled(nCalls int, args []string) error {
 			if prgrm.CallStack[0].Operator == nil {
 				// main function
 				mainCall := MakeCall(fn)
-
+				mainCall.FramePointer = prgrm.StackPointer
 				// initializing program resources
 				prgrm.CallStack[0] = mainCall
 
 				// prgrm.Stacks = append(prgrm.Stacks, MakeStack(1024))
-				prgrm.StackPointer = fn.Size
+				prgrm.StackPointer += fn.Size
 
 				// feeding os.Args
 				if osPkg, err := PROGRAM.SelectPackage(OS_PKG); err == nil {

--- a/cxgo/actions/declarations.go
+++ b/cxgo/actions/declarations.go
@@ -178,7 +178,7 @@ func DeclareImport (ident string, currentFile string, lineNo int) {
 					PRGRM.CurrentPackage = pkg
 
 					if ident == "aff" {
-						AffordanceStructs(imp)
+						AffordanceStructs(imp, currentFile, lineNo)
 					}
 				} else {
 					println(CompilationError(currentFile, lineNo), err.Error())
@@ -191,6 +191,9 @@ func DeclareImport (ident string, currentFile string, lineNo int) {
 }
 
 func DeclareLocal (declarator *CXArgument, declaration_specifiers *CXArgument, initializer []*CXExpression, doesInitialize bool) []*CXExpression {
+	if FoundCompileErrors {
+		return nil
+	}
 	if doesInitialize {
 		declaration_specifiers.IsLocalDeclaration = true
 
@@ -316,13 +319,13 @@ func DeclarationSpecifiersBasic(typ int) *CXArgument {
 	return DeclarationSpecifiers(arg, 0, DECL_BASIC)
 }
 
-func DeclarationSpecifiersStruct (ident string, pkgName string, isExternal bool) *CXArgument {
+func DeclarationSpecifiersStruct (ident string, pkgName string, isExternal bool, currentFile string, lineNo int) *CXArgument {
 	if isExternal {
 		// custom type in an imported package
 		if pkg, err := PRGRM.GetCurrentPackage(); err == nil {
 			if imp, err := pkg.GetImport(pkgName); err == nil {
 				if strct, err := PRGRM.GetStruct(ident, imp.Name); err == nil {
-					arg := MakeArgument("", CurrentFile, LineNo)
+					arg := MakeArgument("", currentFile, lineNo)
 					arg.Type = TYPE_CUSTOM
 					arg.CustomType = strct
 					arg.Size = strct.Size
@@ -333,7 +336,7 @@ func DeclarationSpecifiersStruct (ident string, pkgName string, isExternal bool)
 
 					return arg
 				} else {
-					println(CompilationError(CurrentFile, LineNo), err.Error())
+					println(CompilationError(currentFile, lineNo), err.Error())
 					return nil
 				}
 			} else {
@@ -346,7 +349,7 @@ func DeclarationSpecifiersStruct (ident string, pkgName string, isExternal bool)
 		// custom type in the current package
 		if pkg, err := PRGRM.GetCurrentPackage(); err == nil {
 			if strct, err := PRGRM.GetStruct(ident, pkg.Name); err == nil {
-				arg := MakeArgument("", CurrentFile, LineNo)
+				arg := MakeArgument("", currentFile, lineNo)
 				arg.Type = TYPE_CUSTOM
 				arg.DeclarationSpecifiers = append(arg.DeclarationSpecifiers, DECL_STRUCT)
 				arg.CustomType = strct
@@ -356,7 +359,8 @@ func DeclarationSpecifiersStruct (ident string, pkgName string, isExternal bool)
 
 				return arg
 			} else {
-				panic("type '" + ident + "' does not exist")
+				println(CompilationError(currentFile, lineNo), err.Error())
+				return nil
 			}
 		} else {
 			panic(err)

--- a/cxgo/actions/functions.go
+++ b/cxgo/actions/functions.go
@@ -605,7 +605,8 @@ func ProcessMethodCall (expr *CXExpression, symbols *map[string]*CXArgument, off
 		} else {
 			if out != nil {
 				if argOut, found := (*symbols)[out.Package.Name+"."+out.Name]; !found {
-					panic("")
+					println(CompilationError(out.FileName, out.FileLine), fmt.Sprintf("'%s.%s' not found", out.Package.Name, out.Name))
+					return
 				} else {
 					// then we found an output
 					if len(out.Fields) > 0 {

--- a/cxgo/actions/misc.go
+++ b/cxgo/actions/misc.go
@@ -229,7 +229,7 @@ func StructLiteralFields (ident string) *CXExpression {
 	}
 }
 
-func AffordanceStructs (pkg *CXPackage) {
+func AffordanceStructs (pkg *CXPackage, currentFile string, lineNo int) {
 	// Argument type
 	argStrct := MakeStruct("Argument")
 	// argStrct.Size = GetArgSize(TYPE_STR) + GetArgSize(TYPE_STR)
@@ -324,7 +324,7 @@ func AffordanceStructs (pkg *CXPackage) {
 	prgrmFldFreeHeap.TotalSize = GetArgSize(TYPE_I64)
 
 	// prgrmFldCaller := MakeField("Caller", TYPE_CUSTOM, "", 0)
-	prgrmFldCaller := DeclarationSpecifiersStruct(callStrct.Name, callStrct.Package.Name, false)
+	prgrmFldCaller := DeclarationSpecifiersStruct(callStrct.Name, callStrct.Package.Name, false, currentFile, lineNo)
 	prgrmFldCaller.Name = "Caller"
 
 	prgrmStrct.AddField(prgrmFldCallCounter)

--- a/cxgo/cxgo.nex
+++ b/cxgo/cxgo.nex
@@ -738,6 +738,7 @@ func main () {
 	}
 
 	var prePkg *CXPackage
+	parseErrors := 0
 	
 	if len(sourceCode) > 0 {
 		// we need to traverse the elements by hierarchy
@@ -967,11 +968,14 @@ func main () {
 			if len(fileNames) > 0 {
 				cxgo0.CurrentFileName = fileNames[i]
 			}
-			cxgo0.Parse(source)
+			parseErrors += cxgo0.Parse(source)
 		}
 	}
 
 	PRGRM = cxgo0.PRGRM0
+	if FoundCompileErrors || parseErrors > 0 {
+		os.Exit(CX_COMPILATION_ERROR)
+	}
 
 	if osPkg, err := PRGRM.GetPackage(OS_PKG); err == nil {
 
@@ -985,7 +989,7 @@ func main () {
 
 		DeclareGlobalInPackage(osPkg, arg0, arg1, nil, false)
 	}
-	parseErrors := 0
+
 	// parsing all source code files
 	for i, source := range sourceCodeCopy {
 		source = source + "\n"

--- a/cxgo/cxgo.y
+++ b/cxgo/cxgo.y
@@ -444,15 +444,15 @@ declaration_specifiers:
                 }
         |       IDENTIFIER
                 {
-			$$ = DeclarationSpecifiersStruct($1, "", false)
+			$$ = DeclarationSpecifiersStruct($1, "", false, CurrentFile, LineNo)
                 }
         |       IDENTIFIER PERIOD IDENTIFIER
                 {
-			$$ = DeclarationSpecifiersStruct($3, $1, true)
+			$$ = DeclarationSpecifiersStruct($3, $1, true, CurrentFile, LineNo)
                 }
 	|       type_specifier PERIOD IDENTIFIER
                 {
-			$$ = DeclarationSpecifiersStruct($3, TypeNames[$1], true)
+			$$ = DeclarationSpecifiersStruct($3, TypeNames[$1], true, CurrentFile, LineNo)
                 }
         /* |       package_identifier */
         /*         { */

--- a/cxgo/cxgo0/cxgo0.y
+++ b/cxgo/cxgo0/cxgo0.y
@@ -528,7 +528,7 @@ declaration_specifiers:
                 }
         |       IDENTIFIER
                 {
-			$$ = DeclarationSpecifiersStruct($1, "", false)
+			$$ = DeclarationSpecifiersStruct($1, "", false, CurrentFileName, lineNo)
 			// // custom type in the current package
 			// if pkg, err := PRGRM0.GetCurrentPackage(); err == nil {
 			// 	if strct, err := PRGRM0.GetStruct($1, pkg.Name); err == nil {
@@ -552,7 +552,7 @@ declaration_specifiers:
                 }
         |       IDENTIFIER PERIOD IDENTIFIER
                 {
-			$$ = DeclarationSpecifiersStruct($3, $1, true)
+			$$ = DeclarationSpecifiersStruct($3, $1, true, CurrentFileName, lineNo)
 			// // custom type in an imported package
 			// if pkg, err := PRGRM0.GetCurrentPackage(); err == nil {
 			// 	if imp, err := pkg.GetImport($1); err == nil {
@@ -595,7 +595,7 @@ declaration_specifiers:
                 }
 	|       type_specifier PERIOD IDENTIFIER
 		{
-			$$ = DeclarationSpecifiersStruct($3, TypeNames[$1], true)
+			$$ = DeclarationSpecifiersStruct($3, TypeNames[$1], true, CurrentFileName, lineNo)
 		}
 		/* type_specifier declaration_specifiers */
 	/* |       type_specifier */

--- a/tests/main.cx
+++ b/tests/main.cx
@@ -397,7 +397,7 @@ func main ()() {
 	runTestEx("issue-154.cx", cx.SUCCESS, "Cx stack overflow when appending to a slice passed by address", TEST_ISSUE, 0)
 	runTestEx("issue-155.cx", cx.COMPILATION_ERROR, "Panic when trying to assign return value of a function returning void", TEST_ISSUE, 0)
 	runTestEx("issue-156.cx", cx.COMPILATION_ERROR, "Panic when using a function declared in another package without importing the package", TEST_ISSUE, 0)
-	runTestEx("issue-157.cx", cx.SUCCESS, "Cx memory stomped", TEST_ISSUE, 0)
+	runTest("issue-157.cx", cx.SUCCESS, "Cx memory stomped")
 	runTest("issue-158.cx", cx.SUCCESS, "Invalid offset calculation of non literal strings when appended to a slice")
 
 	var end i64

--- a/tests/main.cx
+++ b/tests/main.cx
@@ -396,7 +396,7 @@ func main ()() {
 	runTestEx("issue-153.cx", cx.SUCCESS, "Panic in when assigning an empty initializer list to a []i32 variable", TEST_ISSUE, 0)
 	runTestEx("issue-154.cx", cx.SUCCESS, "Cx stack overflow when appending to a slice passed by address", TEST_ISSUE, 0)
 	runTestEx("issue-155.cx", cx.COMPILATION_ERROR, "Panic when trying to assign return value of a function returning void", TEST_ISSUE, 0)
-	runTestEx("issue-156.cx", cx.COMPILATION_ERROR, "Panic when using a function declared in another package without importing the package", TEST_ISSUE, 0)
+	runTest("issue-156.cx", cx.COMPILATION_ERROR, "Panic when using a function declared in another package without importing the package")
 	runTest("issue-157.cx", cx.SUCCESS, "Cx memory stomped")
 	runTest("issue-158.cx", cx.SUCCESS, "Invalid offset calculation of non literal strings when appended to a slice")
 

--- a/tests/main.cx
+++ b/tests/main.cx
@@ -390,7 +390,7 @@ func main ()() {
 	runTest("issue-131.cx", cx.SUCCESS, "Problem with struct literals in short variable declarations")
 	runTestEx("issue-132.cx", cx.SUCCESS, "Panic when using the return value of a function in a short declaration", TEST_ISSUE, 0)
 	runTestEx("issue-133.cx", cx.COMPILATION_ERROR, "Panic when inserting a new line in a string literal", TEST_ISSUE, 0)
-	runTestEx("issue-134.cx", cx.COMPILATION_ERROR, "Panic when declaring a variable of an unknown type", TEST_ISSUE, 0)
+	runTest("issue-134.cx", cx.COMPILATION_ERROR, "Panic when declaring a variable of an unknown type")
 	runTestEx("issue-135.cx", cx.COMPILATION_ERROR, "No compilation error when using arithmetic operators on struct instances", TEST_ISSUE, 0)
 	runTestEx("issue-141.cx", cx.SUCCESS, "Parser gets confused with `2 -2`", TEST_ISSUE, 0)
 	runTestEx("issue-153.cx", cx.SUCCESS, "Panic in when assigning an empty initializer list to a []i32 variable", TEST_ISSUE, 0)

